python/agents/product-catalog-ad-generation/scripts/generate_sample_data.py

```diff
@@ -0,0 +1,313 @@
+
+# Output directories
+OUTPUT_DIR = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), "..", "static", "uploads", "products")
+)
+BRANDING_DIR = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), "..", "static", "uploads", "branding")
+)
+
+# --- User Configuration ---
+# Modify these constants to change the generation parameters
+PRODUCT_DESCRIPTION = (
+    "A department store offering a wide range of products including things like "
+    "running shoes, dutch oven, smart bulb, headphones, and a Christmas tree."
+)
+PRODUCT_COUNT = 5
+# Set LOGO_DESCRIPTION to None to skip logo generation
+LOGO_DESCRIPTION = f"A simple, modern logo with stylized letter for {COMPANY_NAME}"
+
+# --- Gemini API Client ---
+# Client is initialized in main()
+
+# --- Pydantic Models ---
+
+class ProductImagePlan(BaseModel):
+    """Plan for a single product image."""
+    filename: str = Field(
+        ..., description="The filename for the image (e.g., 'acme_widget_v1.png')."
+    )
+    image_prompt: str = Field(..., description="The prompt to generate the image.")
+
+class ProductPlanResponse(BaseModel):
+    """The overall response schema for the product plan generation."""
+    products: List[ProductImagePlan]
+
+# --- Core Functions ---
+
+async def generate_plan(
+    client: genai.Client, company_name: str, description: str, count: int
+) -> Optional[List[ProductImagePlan]]:
+    """Generates a plan for creating product images."""
+    prompt = f"""
+    You are an expert creative director. Create a plan to generate {count}
+    high-quality product images for a company named "{company_name}".
+
+    The product description is: "{description}".
+
+    For each product, provide:
+    1. A unique, descriptive filename (ending in .png).
+    2. A detailed, high-quality image generation prompt. The prompt should describe a professional product shot, photorealistic, 8k resolution.
+       The prompt MUST emphasize that the product is isolated on a clean, solid white background, with studio lighting, and no other objects or props in the frame.
+       Focus on the visual details of the product based on the description. The full view of the product should be the main focus of the image.
+
+       CRITICAL: The prompt MUST include instructions on how to naturally incorporate the company logo into the product or its packaging.
+       The logo integration should be subtle, realistic, and match the product's style and material.
+       Examples:
+       - A branded tag on clothing.
+       - An embossed logo on a leather good.
+       - A printed logo on a ceramic mug.
+       - A branded ornament on a Christmas tree.
+       - A sticker or label on a tech gadget.
+       The goal is for the branding to look like a natural part of the physical product.
+
+    Output a JSON object with a 'products' key containing a list of these plans.
+    """
+
+    try:
+        print("\n🤖 Generating product plan with Gemini...")
```

Comment:
Throughout the code, sometimes logging was used to output a message and other times a print. Can we be consistent and just use one (preferably logging)?